### PR TITLE
Update the docs about qlpacks

### PR DESCRIPTION
**Pull request by aeisenberg**
This is a first pass to fix obvious holes and outdated information, but
we should rethink these docs completely.

### CODE REVIEW

**Review by shati-patel — COMMENTED** [review]
You're right that this page probably needs a full rework, but good idea to patch up some obvious inconsistencies 😅 

I don't know too much about the new QL pack structure, but I've left a few small comments (to hopefully get the PR checks passing at least!)

File: docs/codeql/codeql-cli/about-ql-packs.rst [hunk 1]
```diff
@@ -104,7 +104,18 @@ The following properties are supported in ``qlpack.yml`` files.
...
      - Packs with upgrades
      - The path to a directory within the pack that contains upgrade scripts, defined relative to the pack directory. The ``database upgrade`` action uses these scripts to update databases that were created by an older version of an extractor so they're compatible with the current extractor (see `Upgrade scripts for a language <#upgrade-scripts-for-a-language>`__ below.)
+   * - ``dependencies``
+     - .. code-block:: yaml
 
+          dependencies:
+            codeql/cpp-all: ^0.0.2
+
+     - Packs that define CodeQL package dependencies on other packs
+     - A map from pack references to the semantic version range that is compatible with this pack.
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply docs/codeql/codeql-cli/about-ql-packs.rst:114]: Good point.

```suggestion
   * - ``dependencies``
     - .. code-block:: yaml

          dependencies:
            codeql/cpp-all: ^0.0.2

     - Packs that define CodeQL package dependencies on other packs
     - A map from pack references to the semantic version range that is compatible with this pack. Supported from CLI version v2.6.0 and onwards.
```

File: docs/codeql/codeql-cli/about-ql-packs.rst [hunk 2]
```diff
@@ -104,7 +104,18 @@ The following properties are supported in ``qlpack.yml`` files.
...
+   * - ``defaultSuiteFile``
+     - ``defaultSuiteFile: cpp-code-scanning.qls``
+     - Packs that export a set of default queries to run
+     - The path to a query suite file containing all of the queries that are run by default when this pack is passed to the ``codeql database analyze` command.
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply docs/codeql/codeql-cli/about-ql-packs.rst:118]:
```suggestion
     - The path to a query suite file containing all of the queries that are run by default when this pack is passed to the ``codeql database analyze`` command. Supported from CLI version v2.6.0 and onwards.
```

File: docs/codeql/codeql-cli/about-ql-packs.rst [hunk 3]
```diff
@@ -166,48 +177,73 @@ For more information about running tests, see ":doc:`Testing custom queries
...
-   name: codeql-cpp-tests
+   name: codeql/cpp-tests
    version: 0.0.0
-   libraryPathDependencies: codeql/cpp-all
+   dependencies:
+    codeql/cpp-all: "*"
+    codeql/cpp-queries: "*"
 
 Notice that, unlike the example QL pack for custom tests, this file does not define
 an ``extractor`` or ``tests`` property. These properties have been added to
```
Comment (aeisenberg) [inline docs/codeql/codeql-cli/about-ql-packs.rst:249]:
```suggestion
```

**Comment by aeisenberg** [comment]
Thank you for your detailed review!

File: docs/codeql/codeql-cli/about-ql-packs.rst [hunk 4]
```diff
@@ -166,51 +177,74 @@ For more information about running tests, see ":doc:`Testing custom queries
...
-   libraryPathDependencies: codeql/cpp-all
+   dependencies:
+    codeql/cpp-all: "*"
+    codeql/cpp-queries: "*"
 
-Notice that, unlike the example QL pack for custom tests, this file does not define
-an ``extractor`` or ``tests`` property. These properties have been added to
 the QL pack file since the release of CodeQL CLI 2.0.1.
 They haven't been added yet to ensure compatibility for LGTM Enterprise users.
 After the next release of LGTM Enterprise, these files can be updated.
```
Comment (shati-patel) [inline docs/codeql/codeql-cli/about-ql-packs.rst:250]: These 3 lines can also go! 💥 
```suggestion
```

**Review by shati-patel — APPROVED** [review]
One more tiny thing, then LGTM! 📦🚢

**Review by shati-patel — APPROVED** [review]
Re-approving! 👍🏽

**Pull request merged**